msal/src/Microsoft.Identity.Client/Internal/Requests/NonInteractive/UsernamePasswordRequest.cs

```diff
@@ -122,9 +122,17 @@ private async Task UpdateUsernameAsync()
 
         protected override void SetAdditionalRequestParameters(OAuth2Client client)
         {
-            client.AddBodyParameter(OAuth2Parameter.GrantType, OAuth2GrantType.Password);
-            client.AddBodyParameter(OAuth2Parameter.Username, this.usernamePasswordInput.UserName);
-            client.AddBodyParameter(OAuth2Parameter.Password, new string(this.usernamePasswordInput.PasswordToCharArray()));
+            if (this.userAssertion != null)
```

Comment:
I remember you saying this condition is hit only when the account is managed? If so, could you add a comment to make it clearer?